src/app/libp2p_helper/src/libp2p_helper/main.go

```diff
@@ -675,40 +676,52 @@ type incomingMsgUpcall struct {
 }
 
 func handleStreamReads(app *app, stream net.Stream, idx int) {
+	// create buffer stream for non-blocking read
+	r := bufio.NewReader(stream)
+
+	var msgBytes []byte
 	go func() {
 		defer func() {
 			_ = stream.Close()
 		}()
-
-		buf := make([]byte, 4096)
 		for {
-			len, err := stream.Read(buf)
-
-			if len != 0 {
-				app.writeMsg(incomingMsgUpcall{
-					Upcall:    "incomingStreamMsg",
-					Data:      codaEncode(buf[:len]),
+			length, err := readLEB128ToUint64(r)
+			if err == io.EOF {
+				continue
```

Comment:
Quick Go reader question: once the EOF is returned, will the next read block and wait for data rather than returning another EOF?

Reply:
I don't think so, I think it keep reading EOF until there's more data on the stream, but I'll double check